### PR TITLE
Shopware 993 compatibility with shopware 5.6

### PR DESCRIPTION
**Pull request by Skistims**
# Pull Request Template

## Description

SW 5.6 compatibility

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md

### CODE REVIEW

File: src/Backend/SgateShopgatePlugin/Components/Favorites.php [hunk 1]
```diff
@@ -399,4 +418,23 @@ private function removeProductFromWishList($articles, $userId)
 
         return true;
     }
+
+    /**
+     * check for product exists in active category
+     *
+     * @param $productId
```
Comment (andre-kraus) [inline src/Backend/SgateShopgatePlugin/Components/Favorites.php]: type missing in phpdoc + please capitalize the description text of the phpdoc

File: src/Backend/SgateShopgatePlugin/Components/User.php [hunk 2]
```diff
@@ -165,37 +165,52 @@ public function loginUser($request, $httpResponse)
     public function getUser($request)
     {
         try {
-            $decoded    = $this->webCheckoutHelper->getJWT($request->getCookie('token'));
-            $customerId = $decoded['customer_id'];
+            $decoded = $this->webCheckoutHelper->getJWT($request->getCookie('token'));
 
-            $sql = ' SELECT password, email FROM s_user WHERE customernumber = ? AND active=1 AND (lockeduntil < now() OR lockeduntil IS NULL) ';
-            $user = Shopware()->Db()->fetchRow($sql, array($customerId)) ? : array();
+            if ($decoded['error']) {
```
Comment (andre-kraus) [inline src/Backend/SgateShopgatePlugin/Components/User.php]: isset(), or if you don't want to fix all the $decoded related stuff, just add 'error' => false as default to the getJWT method and do $decoded['error'] === true

File: src/Backend/SgateShopgatePlugin/Components/User.php [hunk 3]
```diff
@@ -217,6 +232,13 @@ public function updateUser($request)
             $params  = $this->webCheckoutHelper->getJsonParams($request);
             $decoded = $this->webCheckoutHelper->getJWT($params['token']);
 
+            if ($decoded['error']) {
```
Comment (andre-kraus) [inline src/Backend/SgateShopgatePlugin/Components/User.php]: isset()

File: src/Backend/SgateShopgatePlugin/Components/User.php [hunk 4]
```diff
@@ -249,8 +271,15 @@ public function updateUserEmail($request)
         );
 
         try {
-            $params   = $this->webCheckoutHelper->getJsonParams($request);
-            $decoded  = $this->webCheckoutHelper->getJWT($params['token']);
+            $params  = $this->webCheckoutHelper->getJsonParams($request);
+            $decoded = $this->webCheckoutHelper->getJWT($params['token']);
+
+            if ($decoded['error']) {
```
Comment (andre-kraus) [inline src/Backend/SgateShopgatePlugin/Components/User.php]: isset()

File: src/Backend/SgateShopgatePlugin/Components/User.php [hunk 5]
```diff
@@ -293,8 +322,14 @@ public function updateUserPassword($request)
             'message' => ''
         );
 
-        $params   = $this->webCheckoutHelper->getJsonParams($request);
-        $decoded  = $this->webCheckoutHelper->getJWT($params['token']);
+        $params  = $this->webCheckoutHelper->getJsonParams($request);
+        $decoded = $this->webCheckoutHelper->getJWT($params['token']);
+
+        if ($decoded['error']) {
```
Comment (andre-kraus) [inline src/Backend/SgateShopgatePlugin/Components/User.php]: isset()

File: src/Backend/SgateShopgatePlugin/Components/Address.php [hunk 6]
```diff
@@ -63,9 +63,13 @@ public function __construct()
      */
     public function getAddressesAction($request)
     {
-        $decoded  = $this->webCheckoutHelper->getJWT($request->getCookie('token'));
-        $customer = $this->webCheckoutHelper->getCustomer($decoded['customer_id']);
+        $decoded = $this->webCheckoutHelper->getJWT($request->getCookie('token'));
```
[the remark this answers is not in the document]
Comment (Skistims) [reply src/Backend/SgateShopgatePlugin/Components/Address.php]: I don't understand what you mean. This is a single line of code

File: src/Backend/SgateShopgatePlugin/Components/Favorites.php (hunk 1 above)
Comment (Skistims) [reply src/Backend/SgateShopgatePlugin/Components/Favorites.php]: Good point, the wording is a little off too.

File: src/Backend/SgateShopgatePlugin/Components/Address.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (andre-kraus) [reply src/Backend/SgateShopgatePlugin/Components/Address.php]: Yep, my fault and I'm not even color blind. Forget it.

File: src/Backend/SgateShopgatePlugin/Components/Address.php [hunk 7]
```diff
@@ -136,8 +145,13 @@ public function addAddressAction($request)
      */
     public function deleteAddressAction($request)
     {
-        $params            = $this->webCheckoutHelper->getJsonParams($request);
-        $data              = $this->webCheckoutHelper->getJWT($params['token']);
+        $params = $this->webCheckoutHelper->getJsonParams($request);
+        $data   = $this->webCheckoutHelper->getJWT($params['token']);
```
[the remark this answers is not in the document]
Comment (Skistims) [reply src/Backend/SgateShopgatePlugin/Components/Address.php]: this is an expected part of the request, as this it is responsible for the authentication.

**Pull request merged**